### ISSUE TITLE
General Errors in Unity 2019.3

### ISSUE THREAD

**Issue by reinholdsberzins** [Placeholder-Software/FancyFolders#4]
```
Unable to parse file Assets/Plugins/FancyFolders/Fancy Folders Settings.asset: [Parser Failure at line 1815: Expect ':' between key and value within mapping]

UnityEditor.AssetDatabase:LoadAssetAtPath(String)
FancyFolders.Editor.Settings.FancyFoldersSettings:Reload() (at Assets/Plugins/FancyFolders/Editor/Settings/FancyFoldersSettings.cs:140)
FancyFolders.Editor.Main:Initialize() (at Assets/Plugins/FancyFolders/Editor/Main.cs:32)
FancyFolders.Editor.Main:Update() (at Assets/Plugins/FancyFolders/Editor/Main.cs:136)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()
```
```
Unable to parse file Assets/Plugins/FancyFolders/Fancy Folders Settings.asset: [Parser Failure at line 1906: Expect ':' between key and value within mapping]

UnityEditor.AssetDatabase:LoadAssetAtPath(String)
FancyFolders.Editor.Settings.FancyFoldersSettings:Reload() (at Assets/Plugins/FancyFolders/Editor/Settings/FancyFoldersSettings.cs:140)
FancyFolders.Editor.Main:Initialize() (at Assets/Plugins/FancyFolders/Editor/Main.cs:32)
FancyFolders.Editor.Main:Update() (at Assets/Plugins/FancyFolders/Editor/Main.cs:136)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()
```

**Comment by martindevans** [comment]
This is a serialized Unity asset so a syntax error in this file is very odd! Did you upgrade this project to a newer version of unity, or do anything else that could cause a serialization issue in Unity (e.g. encountered an editor crash).

Could you send me a copy of the file which is referenced in the error message (`Assets/Plugins/FancyFolders/Fancy Folders Settings.asset`)? Either share it here, or if you'd rather keep it private (it contains the names of files in your project) send it to `martin@placeholder-software.co.uk`.

**Comment by reinholdsberzins** [comment]
Here you go
[Fancy Folders Settings.zip](https://github.com/Placeholder-Software/FancyFolders/files/4023545/Fancy.Folders.Settings.zip)

Could there be an issue because the FancyFolders is sitting in Plugins/ folder?

**Comment by martindevans** [comment]
I just checked that file and it's a valid asset file. Was this a transient error (i.e. it only happens occasionally) or does it keep happening until you delete the file?

Edit: I just encountered this issue when upgrading the project to `2019.3` to investigate the other issues.

> Could there be an issue because the FancyFolders is sitting in Plugins/ folder?

That shouldn't be an issue, `Plugins` just separates the compilation into separate steps so that changing your game scripts does not need to recompile everything inside `Plugins`.

**Comment by martindevans** [comment]
I think I've fixed this, it's pending review now.

If you'd like to try the fix out, replace the contents of `ProjectBrowserRenderer.cs` with [this](https://gist.github.com/martindevans/47d7c78b339966eb83a2ca700e32d51d).

**Comment by reinholdsberzins** [comment]
Hi!
 Yes I replaced the CS and it's working perfectly! Thank you! It fixed all the errors and the zoom out error!

**Comment by martindevans** [comment]
Fantastic, thanks for confirming that! We'll have that published to the asset store soon.

**Comment by reinholdsberzins** [comment]
Now when I'm creating the new folder or file its ok, but I just now tried to create new volume for the camera and got these.
![image](https://user-images.githubusercontent.com/24436198/72823776-e2636f00-3c7c-11ea-975e-e6cb68f9b3e1.png)
I'm not sure if this is happening in standard build but it is in UniversalRP.

**Comment by martindevans** [comment]
Could you send me the asset file again (`Assets/Plugins/FancyFolders/Fancy Folders Settings.asset`). Thanks.

**Comment by reinholdsberzins** [comment]
[Fancy Folders Settings.zip](https://github.com/Placeholder-Software/FancyFolders/files/4093178/Fancy.Folders.Settings.zip) Here you go. I've tried on two different Unity project, both are 2019.3 so I'm 99.99% confident it's related to Unity rather than my project.

**Comment by martindevans** [comment]
I think you're correct, it is some kind of Unity issue. The file you sent me is perfectly valid YAML (the format of Unity asset files) but the Unity parser is failing to parse it. If you delete the settings file (while the editor is closed) does this issue persist?

**Comment by reinholdsberzins** [comment]
Wow this seems to fix it! I guess it wants to generate a new settings file not have one one included in package, even if it's valid! :-) So far, no errors, but I will keep you posted.

**Comment by martindevans** [comment]
Thanks for confirming that.

We've just released an update (1.0.1) on the asset store which has the changes mentioned in this thread, as well as some additional hardening around the settings file handling. I'll close this issue for now since it appears to be fixed, but feel free to re-open it or open another one if the issue comes back.
